src/extension/sldService/src/main/java/org/geoserver/sldservice/rest/ImageReader.java

```diff
@@ -134,15 +142,26 @@ public ImageReader invoke() throws IOException, TransformException, FactoryExcep
             readGeometry = new GridGeometry2D(reducedRange, readGeometry.getEnvelope());
         }
 
-        if (!readGeometry.equals(originalGeometry)) {
+        // if there is a filter, set it back as it might have been simplified (e.g., env var
+        // expansion)
+        if (readFilter != null) {
             readParameters =
                     CoverageUtils.mergeParameter(
                             parameterDescriptors,
                             readParameters,
-                            readGeometry,
-                            AbstractGridFormat.READ_GRIDGEOMETRY2D.getName().getCode());
+                            readFilter,
+                            ImageMosaicFormat.FILTER.getName().getCode());
         }
 
+        // if (!readGeometry.equals(originalGeometry)) {
```

Comment:
is this intentionally left commented to explain the use case or is it an oversight instead?

Reply:
Oversight, will fix